bin/all_sky_search/pycbc_template_kde_calc

```diff
@@ -161,17 +163,39 @@ with h5py.File(args.template_file, "r") as f_src:
 temp_samples = numpy.vstack((template_pars)).T
 
 
+# Rejection Sampling
+if 'mchirp' not in args.fit_param:
+    raise ValueError("mchirp does not exist in args.fit_param")
+mchirp_index = args.fit_param.index('mchirp')
+if args.mchirp_downsample_power is not None:
+    power_vals = template_pars[mchirp_index] ** args.mchirp_downsample_power
+    f_dest.attrs['mchirp_downsample_power'] = args.mchirp_downsample_power
+    probabilities = power_vals / numpy.max(power_vals)
+    rand_nums = numpy.random.uniform(0, 1, len(mass1))
+    ind = rand_nums < probabilities
+    new_mchirp = template_pars[mchirp_index][ind]
+    additional_params = numpy.delete(temp_samples, mchirp_index, axis=1)[ind]
+    new_temp_samples = numpy.column_stack((new_mchirp, additional_params))
+
+
 if args.make_template_kde:
     logging.info('Starting optimization of template KDE parameters')
-    optbw, optalpha = optimizedparam(temp_samples, alphagrid=args.alpha_grid,
+    optbw, optalpha = optimizedparam(new_temp_samples, alphagrid=args.alpha_grid,
```

Comment:
This will fail if we are not doing the downsampling. I would call the result of the downsampling something like kde_train_samples, then if we do _not_ do the downsampling you can set kde_train_samples = temp_samples.

Reply:
(since 'new_temp_samples' is rather vague - not clear what is 'new' about them)